### PR TITLE
Allow save image remove-signatures

### PR DESCRIPTION
**Pull request by QiWang19**
remove signatures in podman save since the image formats do not support signatures
Close: #7659 

Signed-off-by: Qi Wang <qiwan@redhat.com>

### CODE REVIEW

File: docs/source/markdown/podman-save.1.md [hunk 1]
```diff
@@ -48,6 +48,10 @@ Allow for creating archives with more than one image.  Additional names will be
 
 Suppress the output
 
+**--remove-signatures**
+
+Do not save signatures, if any, from source image. This is necessary when saving a signed image to a format which does not support signatures. The default is **true**.
```
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-save.1.md:53]:
```suggestion
Do not save signatures, if any, from the source image. This is necessary when saving a signed image to a format that does not support signatures. The default is **true**.
```

**Comment by TomSweeneyRedHat** [comment]
Tests and bash completions?

**Comment by QiWang19** [comment]
bash completions added. I am figuring out how to test this.

**Review by jwhonce — CHANGES_REQUESTED** [review]
@QiWang19 You have new option coded as available to remote users but not coded in the tunnel or bindings packages

File: test/e2e/save_test.go [hunk 2]
```diff
@@ -116,6 +119,48 @@ var _ = Describe("Podman save", func() {
...
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+
+		session = podmanTest.Podman([]string{"pull", "--tls-verify=false", "localhost:5000/alpine"})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+
+		outfile := filepath.Join(podmanTest.TempDir, "temp.tar")
+		save := podmanTest.Podman([]string{"save", "remove-signatures=true", "-o", outfile, "localhost:5000/alpine"})
+		save.WaitWithDefaultTimeout()
```
Comment (jwhonce) [inline test/e2e/save_test.go:160]: Can you verify that signatures are not included?

Comment (QiWang19) [reply test/e2e/save_test.go:179]: I haven't found an easy way to verify that. Needs config the registries.d/default.yaml to and gpg keys to verify the signature. Updated the test but not sure if that can work.

File: test/e2e/save_test.go [hunk 3]
```diff
@@ -116,6 +120,71 @@ var _ = Describe("Podman save", func() {
 		Expect(save).To(ExitWithError())
 	})
 
+	It("podman save remove signature", func() {
+		SkipIfRemote("podman-remote doesn't suppport tls-verify option")
```
Comment (rhatdan) [inline test/e2e/save_test.go:124]: Podman remote now supports tls-verify.

**Review by rhatdan — APPROVED** [review]
When would a user ever specify --remove-signatures=false?

**Comment by QiWang19** [comment]
> When would a user ever specify --remove-signatures=false?

The user will use  --remove-signatures=false, current image format doesn't support pushing signatures to it.

File: test/e2e/save_test.go (hunk 3 above)
Comment (QiWang19) [reply test/e2e/save_test.go]: podman remove push --tls-verify works, but still see `unknown flag: --tls-verify` with podman pull,
in the test **int remote ubuntu-19 root host**
```
[+2546s] Running: /var/tmp/go/src/github.com/containers/podman/bin/podman-remote --remote --url unix:/run/podman/podman-1695fe65d29f17c37c0cd8754dbdd796e29036e2adea66ec67ab760001892f06.sock --remote --url unix:/run/podman/podman-1695fe65d29f17c37c0cd8754dbdd796e29036e2adea66ec67ab760001892f06.sock pull --tls-verify=false --signature-policy=sign/policy.json localhost:5000/alpine
[+2546s] Error: unknown flag: --tls-verify
```

Comment (rhatdan) [reply test/e2e/save_test.go]: I opened a PR to fix this.

https://github.com/containers/podman/pull/8042

**Comment by rhatdan** [comment]
@QiWang19 Shouldn't the code do this automatically, IE Does the code know which image stores support signatures?  Would the user just get an error telling them to remove signatures and now they need to run the command again with the flag?  Does not seem like a good UI experience?

**Comment by QiWang19** [comment]
On the suggestion under the issue https://github.com/containers/podman/issues/7659#issuecomment-694121445, the remove-signatures defaults to true and avoids the user from getting the error.

**Comment by rhatdan** [comment]
@QiWang19 Sure,  but under what circumstance would the user ever set this to False.  When would the user not want to remove-signatures?  Bottom line is there a situation where a user would ever type in 
--remove-signatures=false or true?  
If we want signatures to be pushed to a registry?

**Comment by QiWang19** [comment]
If we want signatures to be pushed to a registry, `podman push` also has `remove-signatures` defaults to false. This patch only applies to `podman save`, current format (oci-archive, oci-dir, docker-dir) doesn't support save with signatures.

**Comment by rhatdan** [comment]
When would I ever do a 
podman save --remove-signatures=false?

**Comment by QiWang19** [comment]
> When would I ever do a
> podman save --remove-signatures=false?

Remove the option and remove the signature in the code, the current image format doesn't support saving signatures.

**Comment by rhatdan** [comment]
LGTM

**Comment by TomSweeneyRedHat** [comment]
LGTM
Would like a head nod from @jwhonce before merging as he'd asked for a change.

**Review by jwhonce — APPROVED** [review]

**Pull request merged**